### ISSUE TITLE
[BUG] Not compatible with react-router v6

### ISSUE THREAD

**Issue by changesbyjames** [microsoft/ApplicationInsights-JS#1673]
It's not a huge deal as v6 is currently in beta but there is no exposed concept of `history` anymore so the way that the react plugin works with the history object will be unsupported soon. 

See here for more detail: https://github.com/remix-run/react-router/issues/7476

**Comment by paularmstrong** [comment]
v6 is now published to npm and is the recommended version going forward. They react-router docs and issues are really cryptic about the history module and they don't seem to even expose it in any way. I'm still searching around, but would be helpful to get some more eyes on the lookout for this.

**Comment by afefer** [comment]
There's a [HistoryRouter object](https://github.com/remix-run/react-router/pull/7586) that can be used in place of Router that solves this problem.

**Comment by MironV** [comment]
In v6, `useNavigate` and `useLocation` replace `useHistory` in react-router. It would be best if this package could simply use the hooks provided by react-router rather than having to change BrowserRouter to HistoryRouter, which is essentially a compatibility mode to enable extracting history and could introduce bugs.

**Comment by XzaR90** [comment]
@afefer I don't see any function named createHistory in "react-router-dom": "6.2.1",
How to get this work with v6?

**Comment by afefer** [comment]
> @afefer I don't see any function named createHistory in "react-router-dom": "6.2.1", How to get this work with v6?

It's flagged as unstable in 6.2.1, so you can import it as:
`
import { unstable_HistoryRouter as HistoryRouter } from 'react-router-dom';
`

**Comment by XzaR90** [comment]
hmm may not be a good solution for production then, I will probably wait. Thank you @afefer.

**Comment by XzaR90** [comment]
Can't we just have a custom event that reactplugin is listen to which having the pathname?
Then we can trigger the custom event using an example like this https://stackoverflow.com/a/69785094/3153684?

`const event = new CustomEvent('ai_pathname_change', { pathname: '...' });`

ReactPlugin can have a static method creating that event to make it easier for developer to use it.

@kryalama, @Karlie-777

**Comment by Karlie-777** [comment]
Hi all @changesbyjames @paularmstrong @afefer @MironV  @XzaR90 
for V6, set ``enableAutoRouteTracking: true`` in ApplicationInsights config settings and then you do not need to pass ``history`` to enable auto route tracking. please let me know know if it works.

**Comment by sandves** [comment]
> Hi all @changesbyjames @paularmstrong @afefer @MironV @XzaR90 for V6, set `enableAutoRouteTracking: true` in ApplicationInsights config settings and then you do not need to pass `history` to enable auto route tracking. please let me know know if it works.

`enableAutoRouteTracking: true` does seem to track page views, but it does not capture durations as far as I can see.

**Comment by MSNev** [comment]
Correct, because it has no way to identify how "long" it took to render the page, which fundamentally is what duration is about, where for the initial page this includes how long it took to download the resources up until the page load event.

For a SPA, "generally" all resources are local. If you want to track async (loading views etc) then you would have to use (or do something like) the `startTrackPage` / `stopTrackPage` from the analytics extension
